Add React Conf 2024 Recap post (#6886)
reactjs/react.dev: src/content/blog/2024/05/20/react-conf-2024-recap.md

---
title: "React Conf 2024 Recap"
author: Ricky Hanlon
date: 2024/05/20
description: Last week we hosted React Conf 2024, a two-day conference in Henderson, Nevada where over 800 attendees gathered to discuss the latest in UI engineering. This was our first in-person conference since 2019, and we were thrilled to be able to bring the community together again.
---

May 20, 2024 by [Ricky Hanlon](https://twitter.com/rickhanlonii).

---

<Intro>

Last week we hosted React Conf 2024, a two-day conference in Henderson, Nevada where over TODO attendees gathered to discuss the latest in UI engineering. This was our first in-person conference since 2019, and we were thrilled to be able to bring the community together again.

</Intro>

---

> At React Conf 2024, we announced the React 19 RC, React Native New Architecture Beta, the experimental release of the React Compiler, and much more. The entire [Day One](https://www.youtube.com/watch?v=T8TZQ6k4SLE) and [Day Two](https://www.youtube.com/watch?v=0ckOUBiuxVY) stream is available online. 

Reviewer: ```suggestion
At React Conf 2024, we announced the React 19 RC, React Native New Architecture Beta, the experimental release of the React Compiler, and much more. The entire [Day One](https://www.youtube.com/watch?v=T8TZQ6k4SLE) and [Day Two](https://www.youtube.com/watch?v=0ckOUBiuxVY) streams are available online. 
```

In this post, we'll summarize the talks and announcements from the event.

> ## Day 1 {/*day-1*/}

Reviewer 2: I’d recommend including a link to the keynote somewhere up here, even just an anchor link to the embedded video later in this doc. It’s a long time before the reader can find what you are referencing

Reviewer 2: Same for the day 2 section

To kick off Day 1, Meta CTO [Andrew "Boz" Bosworth](https://www.threads.net/@boztank) shared a welcome message followed by an introduction by [Seth Webster](https://twitter.com/sethwebster) and our MC [Ashley Narcisse](https://twitter.com/_darkfadr).

> In the Day 1 Keynote, [Joe Savona](https://twitter.com/en_JS) shared our goals and vision for React to make it easy for anyone to build great user experiences. [Lauren Tan](https://twitter.com/potetotes) followed with the State of React, which was downloaded over 1 billion times in 2023 and highlighted the work of the React Community to make React, React.

Reviewer 2: This reads like the state of react talk was downloaded that many times

Author: Fixed

> In the day 1 keynote, [Joe Savona](https://twitter.com/en_JS) shared our goals and vision for React to make it easy for anyone to build great user experiences. [Lauren Tan](https://twitter.com/potetotes) followed with a State of React, which was downloaded over 1 billion times in 2023 and highlighted the work of the React community to make React, React.

Reviewer 2: This sounds like the state of react was downloaded a billion times instead of react the npm package

Author: Done

For more, check out these talks from the community later in the conference:

- [Vanilla React](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Ryan Florence](https://twitter.com/ryanflorence)
- [React Rhythm & Blues](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=12728s) by [Lee Robinson](https://twitter.com/leeerob)
- [RedwoodJS, now with React Server Components](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=26815s) by [Amy Dutton](https://twitter.com/selfteachme)
- [Introducing Universal React Server Components in Expo Router](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Evan Bacon](https://twitter.com/Baconbrix)

> Next in the Keynote, [Josh Story](https://twitter.com/joshcstory) and [Andrew Clark](https://twitter.com/acdlite) shared new features [coming in React 19](/blog/2024/04/25/react-19), and announced the React 19 RC is ready for testing in production. Check out all the features in the React 19 [release announcement post](/blog/2024/04/25/react-19), and see these talks for deep dives on the new features:

Reviewer: ```suggestion
Next in the Keynote, [Josh Story](https://twitter.com/joshcstory) and [Andrew Clark](https://twitter.com/acdlite) shared new features [coming in React 19](/blog/2024/04/25/react-19), and announced the React 19 RC which is ready for testing in production. Check out all the features in the React 19 [release announcement post](/blog/2024/04/25/react-19), and see these talks for deep dives on the new features:
```

- [What's new in React 19](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Lydia Hallie](https://twitter.com/lydiahallie)
- [React Unpacked: A Roadmap to React 19](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Sam Selikoff](https://twitter.com/samselikoff)
- [React 19 Deep Dive: Coordinating HTML](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Josh Story](https://twitter.com/joshcstory)
- [Enhancing Forms with React Server Components](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=25280s) by [Aurora Walberg Scharff](https://twitter.com/aurorascharff)
- [React for Two Computers](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Dan Abramov](https://twitter.com/dan_abramov2)
- [And Now You Understand React Server Components](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=11256s) by [Kent C. Dodds](https://twitter.com/kentcdodds)

> Finally, we ended the Keynote with [Joe Savona](https://twitter.com/en_JS), [Sathya Gunasekaran](https://twitter.com/_gsathya), and [Mofei Zhang](https://www.threads.net/@z_mofei) announcing that the React Compiler is now Open Source, and sharing an experimental version of the React Compiler to try out.

Reviewer 3: ```suggestion
Finally, we ended the Keynote with [Joe Savona](https://twitter.com/en_JS), [Sathya Gunasekaran](https://twitter.com/_gsathya), and [Mofei Zhang](https://twitter.com/zmofei) announcing that the React Compiler is now Open Source, and sharing an experimental version of the React Compiler to try out.
```

Author: Done

> For more information on using the Compiler and how it works, check out these talks:

Reviewer 3: ```suggestion
For more information on using the Compiler and how it works, check out [the docs](https://react.dev/learn/react-compiler) as well as these talks:
```

Author: Haha we had the same thought, I already added this 👍

- [Forget About Memo](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=5542s) by [Lauren Tan](https://twitter.com/potetotes)

> - [React Compiler Deep Dive](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=9313s) by [Sathya Gunasekaran](https://twitter.com/_gsathya) and [Mofei Zhang](https://www.threads.net/@z_mofei)

Reviewer 3: ```suggestion
- [React Compiler Deep Dive](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=9313s) by [Sathya Gunasekaran](https://twitter.com/_gsathya) and [Mofei Zhang](https://twitter.com/zmofei)
```

Watch the full day 1 keynote here:

<YouTubeIframe src="https://www.youtube.com/embed/T8TZQ6k4SLE?t=973s" />

## Day 2 {/*day-2*/}

To kick off day 2, [Seth Webster](https://twitter.com/sethwebster) shared a welcome message, followed by a Thank You from [Eli White](https://x.com/Eli_White) and an introduction by our Chief Vibes Officer [Ashley Narcisse](https://twitter.com/_darkfadr).

> In the day 2 keynote, [Nicola Corti](https://twitter.com/cortinico) shared the State of React Native, including 78 million downloads in 2023, and highlighted apps using React Native including 2000+ screens used inside of Meta, and screens from Microsoft in Microsoft Office and the Microsoft Windows Start Menu.

Reviewer 2: The growth of downloaded matters more than the absolute number

Reviewer 2: Do you want to mention that the Marketplace PDP page, written with React Native, is visited more than 2 Billion times a day? We can

Author: Can you add a suggested edit? Not sure what to say here instead.

Reviewer 2: Idk how to do suggested change from mobile, but here is a proposal

> In the day 2 keynote, [Nicola Cortil(https:/ twitter.com/cortinico) shared the State of React Native, including 78 million downloads of React Native in 2023, and highlighted apps using React Native including 2000+ screens used inside of Meta. Facebook Marketplace's product details page, written with React Native, is visited more than 2 billion times per day. Nicola also shared that at Microsoft, part of the Windows Start Menu and some features in almost every Microsoft Office product across mobile and desktop are written with React Native.

- [React Native Q&A](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=27935s) hosted by [Jamon Holmgren](https://twitter.com/jamonholmgren)

## And more... {/*and-more*/}

We also heard talks on accessibility, error reporting, css, and more:

- [Demystifying accessibility in React apps](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=20655s) by [Kateryna Porshnieva](https://twitter.com/krambertech)
- [Pigment CSS, CSS in the server component age](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=21696s) by [Olivier Tassinari](https://twitter.com/olivtassinari)
- [Real-time React Server Components](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=24070s) by [Sunil Pai](https://twitter.com/threepointone)
- [Let's break React Rules](https://www.youtube.com/watch?v=T8TZQ6k4SLE&t=25862s) by [Charlotte Isambert](https://twitter.com/c_isambert)
- [Solve 100% of your errors](https://www.youtube.com/watch?v=0ckOUBiuxVY&t=19881s) by [Ryan Albrecht](https://github.com/ryan953)

## Thank you {/*thank-you*/}

Thank you to all the Staff, Speakers, and Participants who made React Conf 2024 possible. There are too many to list, but we want to thank a few in particular.

> Thank you to [Barbara Markiewicz](https://twitter.com/barbara_markie), the team at [Callstack](https://www.callstack.com/), and our React Team Developer Advocate [Matt Carroll](https://twitter.com/mattcarrollcode) for helping to plan the entire event; and to [Sunny Leggett](https://zeroslopeevents.com/about) and everyone from [Zero Slope](https://zeroslopeevents.com) for helping to organize the event.

Reviewer 3: might be awkward to thank yourself, but we should totally thank you too @rickhanlonii for all the late nights you spent making our slides amazing, providing a ton of super useful feedback, and generally helping out at key moments and within the "boundaries" of things where things could easily slip. Thank you!!

Reviewer 2: Agreed.

Thank you to Rick Hanlon who helped the speakers prepare great talks, made the keynote slides pretty, and helped wherever needed additional attention.

Author: Done <3

Thank you [Ashley Narcisse](https://twitter.com/_darkfadr) for being our MC and Chief Vibes Officer; and to [Michael Chan](https://twitter.com/chantastic) and [Jamon Holmgren](https://twitter.com/jamonholmgren) for hosting the Q&A sessions.

> 

Reviewer 3: ```suggestion

Thank you [Ricky Hanlon](https://twitter.com/rickhanlonii) for providing detailed feedback to speakers on their talks, working on design and animations for the keynote slides, deploying the [React Compiler Playground](https://playground.react.dev), and generally filling in the gaps to sweat the details and make this conference a success.
```

Thank you [Callstack](https://www.callstack.com/) for building the conference website; and to [Kadi Kraman](https://twitter.com/kadikraman) and the [Expo](https://expo.dev/) team for building the conference mobile app.

Thank you to all the sponsors who made the event possible: [Remix](https://remix.run/), [Amazon](https://developer.amazon.com/apps-and-games?cmp=US_2024_05_3P_React-Conf-2024&ch=prtnr&chlast=prtnr&pub=ref&publast=ref&type=org&typelast=org), [MUI](https://mui.com/), [Sentry](https://sentry.io/for/react/?utm_source=sponsored-conf&utm_medium=sponsored-event&utm_campaign=frontend-fy25q2-evergreen&utm_content=logo-reactconf2024-learnmore), [Abbott](https://www.jobs.abbott/software), [Expo](https://expo.dev/), [RedwoodJS](https://redwoodjs.com/), and [Vercel](https://vercel.com).

Thank you to the AV Team for the visuals, stage, and sound; and to the Westin Hotel for hosting us.

Thank you to all the speakers who shared their knowledge and experiences with the community.

> And thank you to everyone who attended in person and online for making React Conf 2024 great.

Reviewer 2: Can you add one more sentence here to make people end this post with a warm fuzzy feeling that matches the vibe of the event?

Maybe something about how inspiring it was to see all the conversations and attendees work together to make React better. Or a deeper thank you to the community and educators and maintainers for supporting React. 

Just something to establish that tone. Maybe you have other ideas

Author: Good idea, but nothing comes that doesn't seem cheesy or forced.

Author: Updated!
